src/platform/telemetry/index.ts

```diff
@@ -141,8 +141,14 @@ const queuedTelemetry: {
  */
 export function sendTelemetryEvent<P extends IEventNamePropertyMapping, E extends keyof P>(
     eventName: E,
-    durationMs?: Record<string, number> | number,
-    properties?: P[E] & { [waitBeforeSending]?: Promise<void> },
+    measures?:
+        | (P[E] extends TelemetryEventInfo<infer R> ? PickType<UnionToIntersection<R>, number> : undefined)
+        | undefined,
+    properties?: P[E] extends TelemetryEventInfo<infer R>
```

Comment:
Same API, but better type inference and we now have type checks for measures.
I.e. everything is strongly typed